src/control/jobs.c

```diff
@@ -404,10 +448,15 @@ int dt_control_add_job(dt_control_t *control, dt_job_queue_t queue_id, _dt_job_t
   pthread_cond_broadcast(&control->cond);
   dt_pthread_mutex_unlock(&control->cond_mutex);
 
+  // dispose of dropped job, if any
+  dt_control_job_set_state(job_for_disposal, DT_JOB_STATE_DISCARDED);
+  dt_control_job_dispose(job_for_disposal);
+
   return 0;
 }
 
 static __thread int threadid = -1;
+static __thread pthread_t pthreadid = -1;
```

Comment:
I don't think this is possible. pthread_t just happens to be an int on Linux, but that is not the case on other platforms.